Clean code

skrakau opened the pull request:
Changed remaining channel names to `ch_*` and tried to make names more consistent.

Made sample variable names for short and long read processes consistent (-> `name`). And changed variable name in downstream processed accordingly (`sample` -> `name`), will be useful when option for pooling will be added (`name` could be either sample name or group name). 

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated
- [ ] If necessary, also make a PR on the [nf-core/mag branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/mag)

d4straub commented on main.nf:
```diff
@@ -595,10 +595,10 @@ if(!params.keep_phix) {
 
         output:
-        set val(name), file("*.fastq.gz") into (trimmed_reads_megahit, trimmed_reads_metabat, trimmed_reads_fastqc, trimmed_sr_spadeshybrid, trimmed_reads_spades, trimmed_reads_centrifuge, trimmed_reads_kraken2, trimmed_reads_bowtie2, trimmed_reads_filtlong)
+        set val(name), file("*.fastq.gz") into (ch_reads_megahit, ch_trimmed_reads_metabat, ch_trimmed_reads_fastqc, ch_short_reads_spadeshybrid, ch_reads_spades, ch_reads_centrifuge, ch_trimmed_reads_kraken2, ch_reads_bowtie2, ch_short_reads_filtlong)
```
Why sometimes with "trimmed" and sometimes without? E.g. `ch_reads_centrifuge`,`ch_reads_megahit` but `ch_trimmed_reads_kraken2`,`ch_trimmed_reads_metabat`?

It does make sense for fastqc, you have currently `ch_raw_reads_fastqc` and `ch_trimmed_reads_fastqc`. But also here, for the sake of standardising, "trimmed" could be omitted and only "raw" could be used if its not trimmed.

Also, it might be actually good to always add "short" or "long" for reads, might make it even more readable and easier to follow.

d4straub commented on main.nf:
```diff
@@ -674,26 +674,26 @@ process fastqc_trimmed {
 
         input:
-        set id, file(lr) from files_long_raw
+        set val(name), file(lr) from ch_long_reads_raw
```
Following the scheme of `ch_raw_reads_fastqc`, `ch_long_reads_raw` could be also named `ch_raw_long_reads`

d4straub commented on main.nf:
```diff
@@ -674,26 +674,26 @@ process fastqc_trimmed {
-        set id, file("${id}_porechop.fastq") into files_porechop
-        set id, file(lr), val("raw") into files_nanoplot_raw
+        set val(name), file("${name}_porechop.fastq") into ch_porechop_reads
+        set val(name), file(lr), val("raw") into ch_raw_reads_nanoplot
```
`ch_raw_reads_nanoplot` misses the "long"

d4straub commented on main.nf:
```diff
@@ -674,26 +674,26 @@ process fastqc_trimmed {
-    pre_files_nanoplot_raw
-        .map { id, lr -> [ id, lr, "raw" ] }
-        .set { files_nanoplot_raw }
+    ch_long_reads_raw
+        .into{ ch_porechop_reads; ch_raw_reads_pre_nanoplot }
+    ch_raw_reads_pre_nanoplot
+        .map { name, lr -> [ name, lr, "raw" ] }
+        .set { ch_raw_reads_nanoplot }
```
"raw" sometimes at the end sometimes in the middle.

d4straub commented on main.nf:
```diff
@@ -759,19 +759,19 @@ process filtlong {
 
     input:
-    set id, file(lr), type from files_nanoplot_raw.mix(files_nanoplot_filtered)
+    set val(name), file(lr), val(type) from ch_raw_reads_nanoplot.mix(ch_reads_filtered_nanoplot)
```
Why not `ch_raw_reads_nanoplot` and `ch_filtered_reads_nanoplot` instead `ch_reads_filtered_nanoplot`

d4straub reviewed (approved):
That's actually great, to bring more consistency! I have a few comments, but all of these are not important at all.

skrakau commented:
Thanks @d4straub ! I thought I leave it as minimally invasive as possible, but you are right! :) I will try to include your suggestions

skrakau commented:
As suggested I added 'short' and 'long', removed the 'trimmed' and improved the order.

In general:
`ch_x_short_reads`: 'x' was already applied on the reads
`ch_short_reads_x:` reads will go to process 'x' next.

Thus there are still a few inconsistencies, i.e. sometimes the name is coming from the last process, and sometimes the name says where the channel is going next. Maybe something for the future...
